File: leetcode/cpp/firstUniqChar.cc
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> counter;
        for(char it : s){
            counter[it] += 1;
        }
        for(int i=0; i<s.size(); i++){
            if(counter[s[i]] == 1) return i;
        }
        return -1;
    }
};

File: leetcode/py/firstUniqChar.py
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        mm = {}
        for it in s:
            if it in mm:
                mm[it] += 1
            else:
                mm[it] = 1
        for idx in range(len(s)):
            if mm[s[idx]] == 1:
                return idx
        return -1
    